commit a8cf509a06cdc42568e446dea8a54a398e1c003b
Author: Benjamin Reed <rangerrick@befunk.com>
Date:   Wed Jul 22 08:47:51 2009 -0400

    Fix compilation on non-Linux: the F_DUPFD_CLOEXEC check was reversed
    
    Signed-off-by: Thiago Macieira <thiago@kde.org>

diff --git a/dbus/dbus-sysdeps-unix.c b/dbus/dbus-sysdeps-unix.c
index f0f1e70..e60e614 100644
--- a/dbus/dbus-sysdeps-unix.c
+++ b/dbus/dbus-sysdeps-unix.c
@@ -3043,7 +3043,7 @@ _dbus_dup(int        fd,
     return -1;
   }
 
-#ifndef F_DUPFD_CLOEXEC
+#ifdef F_DUPFD_CLOEXEC
   if (!cloexec_done)
 #endif
     {
